src/main/java/service/impl/DataProcessorServiceImpl.java

```diff
@@ -0,0 +1,52 @@
+package service.impl;
+
+import db.Storage;
+import java.util.List;
+import model.FruitTransaction;
+import operations.OperationHandler;
+import service.DataProcesorService;
+import service.OperationStrategy;
+
+public class DataProcessorServiceImpl implements DataProcesorService {
+    private final Storage storage;
+    private final OperationStrategy operationStrategy;
+
+    public DataProcessorServiceImpl(Storage storage, OperationStrategy operationStrategy) {
+        this.storage = storage;
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<String> inputLines) {
+        for (String line : inputLines) {
+            FruitTransaction transaction = parseTransaction(line);
+            if (transaction != null) {
+                OperationHandler handler = operationStrategy
+                        .getOperationHandler(transaction.getOperation());
+                if (handler != null) {
+                    handler.execute(storage, transaction);
+                } else {
+                    throw new IllegalArgumentException("Unsupported operation: "
+                        + transaction.getOperation());
+                }
+            }
+        }
+    }
+
+    public FruitTransaction parseTransaction(String line) {
+        String[] tokens = line.split(",");
```

Comment:
I would move it to constant -- psfs(private static final String) SEPARATOR = ","